app/src/main/java/kt/school/starlord/di/PageParser.kt

```diff
@@ -0,0 +1,40 @@
+package kt.school.starlord.di
+
+import kt.school.starlord.entity.Category
+import kt.school.starlord.entity.Subcategory
+
+class PageParser {
+    var allCategories: ArrayList<Category> = ArrayList()
+
+    fun parsePage(loadedPage: String): ArrayList<Category> {
+        /* get from the page all "oneCat" divisions */
+        val allData: Sequence<MatchResult> = Regex("""<h3>(.|\n)*?</div>""").findAll(loadedPage)
```

Comment:
```suggestion
        val allData: Sequence<MatchResult> = """<h3>(.|\n)*?</div>""".toRegex().findAll(loadedPage)
```
исправить во всех местах
https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/to-regex.html